### DISCUSSION TITLE
[3DS] Tweak Touchscreen On 3DS

### DISCUSSION THREAD

**Discussion by Kristopian** [diasurgical/DevilutionX#3654] [Ideas]
On old 3ds, no idea about this on new 3ds

Please make it an option to change stylus control. So that you click on release instead of click on touch. Another solution is to display the game on both screens. Just the menu on the bottom, some shortcuts, or just leave it blank.

**Comment by AJenbo** [comment]
@Kristopian please use Discussions for ideas. Also posting multiple messages, in the style of what people are requestion that you don't, isn't conducive to a good conversation about the subject. I tried improving your message to make it easier to understand. Feel free to correct it if it's not correct.

**Reply by Kristopian** [reply]
> @Kristopian please use Discussions for ideas. Also posting multiple messages, in the style of what people are requestion that you don't, isn't conducive to a good conversation about the subject. I tried improving your message to make it easier to understand. Feel free to correct it if it's not correct.

Thanks, I will try not to capitalize every word.

**Reply by Kristopian** [reply]
Here's a possible touchscreen design. ( >_<)

![3ds Touchscren Idea](https://user-images.githubusercontent.com/91993172/144138930-31dc0f6d-4ed9-4a25-af17-a83b8bb68321.png)

-EDIT Not my New 3DS XL just a google image

**Reply by AJenbo** [reply]
We currently aren't able to output to the lower screen, which makes this out of reach for now.

**Reply by Kristopian** [reply]
> We currently aren't able to output to the lower screen, which makes this out of reach for now.

Got It

**Comment by Kristopian** [comment]
This one Isn't as open, or clean, but it gets the job done

![3DS Touchscreen Idea 2](https://user-images.githubusercontent.com/91993172/144305508-c728712e-62ae-4883-b3e4-20bc421d96bc.png)

I have no Idea how to rip textures, but I currently don't need to.

**Comment by Kristopian** [comment]
I spent some time trying to implement a bottom screen for 3ds but the main thread keeps getting locked up making some things(mainly multiplayer) impossible, do you guys have any idea how to fix this.

**Reply by AJenbo** [reply]
is it related to the sound crash in master? that's a bug in the SDL port for 3DS, we sent a fix but DevkitPro hasn't looked at it yet.

**Reply by Kristopian** [reply]
> is it related to the sound crash in master? that's a bug in the SDL port for 3DS, we sent a fix but DevkitPro hasn't looked at it yet.

I have no idea, but when I try to print some graphics to both screens the main thread keeps locking up making it really hard and even impossible to do some basic things in game(mainly multiplayer and a couple inventory functions).

**Reply by StephenCWills** [reply]
The only thing I've done with the bottom screen so far is setting `SDL_CONSOLEBOTTOM` for debugging and also messing with the code that disables the backlight. However, I don't really understand why using both screens would cause the main thread to lock up so there's probably something I'm missing. Are you using `SDL_DUALSCR`, or are you trying to replace SDL with Citro3D/Citro2D, or what?

**Reply by Kristopian** [reply]
I tried putting the ctr keyboard applet to to the bottom 24/7, and turning on the backlight.

**Reply by StephenCWills** [reply]
That explains it, actually. That's just how the keyboard applet works, and I don't think there's any way around it. I'm personally in favor of replacing the keyboard applet with a software keyboard, which would also help a few other platforms (#2249).

**Reply by Kristopian** [reply]
> That explains it, actually. That's just how the keyboard applet works, and I don't think there's any way around it. I'm personally in favor of replacing the keyboard applet with a software keyboard, which would also help a few other platforms (#2249).

Oh.

**Reply by Kristopian** [reply]
How did I miss that

**Comment by fwannmacher** [comment]
i have an old code (2 years old) where i was toying around rendering the map and the panels in the bottom screen and the game in the top. You can take a look if you wish.
[3ds.txt](https://github.com/diasurgical/devilutionX/files/7897034/3ds.txt)

**Reply by Kristopian** [reply]
Should I use Citra, or keep writing and testing the code on my 3ds.

**Reply by Kristopian** [reply]
I am getting yelled at in other forums for not using Citra, or another 3ds emulator.

**Reply by AJenbo** [reply]
Doing it on your 3DS is fine, but you could probably test things a lot faster if you used an emulator. You likely will also get better debug information. I would use real hardware mostly as a sanity check once every now and then. In the end it's up to you what workflow you are most efficient with, but learn both before rejecting either.

**Reply by Kristopian** [reply]
> Doing it on your 3DS is fine, but you could probably test things a lot faster if you used an emulator. You likely will also get better debug information. I would use real hardware mostly as a sanity check once every now and then. In the end it's up to you what workflow you are most efficient with, but learn both before rejecting either.

I have used Citra but I noticed that things are (slightly) different and I lose a bit of sanity every time I spot something (slightly) off.

**Reply by Kristopian** [reply]
So I feel slightly better using native tools.
